### ISSUE TITLE
ComfyUI comes up with very different images than Diffusers

### ISSUE THREAD

**Issue by komninoschatzipapas** [comfyanonymous/ComfyUI#3507]
I'm using this Python script:
```python
# Use Pytorch 2!
import torch
from diffusers import (
    StableDiffusionPipeline,
    DiffusionPipeline,
    AutoencoderKL,
    UNet2DConditionModel,
    DDPMScheduler,
)
from transformers import CLIPTextModel

# Any model currently on Huggingface Hub.
model_id = "stabilityai/stable-diffusion-xl-base-1.0"
pipeline = DiffusionPipeline.from_pretrained(model_id)

# Optimize!
pipeline.unet = torch.compile(pipeline.unet)
scheduler = DDPMScheduler.from_pretrained(model_id, subfolder="scheduler")

# Remove this if you get an error.
torch.set_float32_matmul_precision("high")

pipeline.to("cuda")
prompts = {
    "car": "car",
}
for shortname, prompt in prompts.items():
    # old prompt: ''
    image = pipeline(
        prompt=prompt,
        negative_prompt="",
        num_inference_steps=35,
        generator=torch.Generator(device="cuda").manual_seed(42),
        width=832,
        height=1216,
        guidance_scale=5,
    ).images[0]
    image.save(f"./{shortname}_nobetas.png", format="PNG")
```
Which yields this image:
![car_nobetas](https://github.com/bghira/SimpleTuner/assets/29049104/a7f60d54-d0da-4ade-b485-74483be4d546)
  
Trying to replicate this in ComfyUI (`719fb2c`):
![CleanShot 2024-05-17 at 21 25 06](https://github.com/bghira/SimpleTuner/assets/29049104/7e62322e-42bc-44b6-bd95-de25dec789e1)

I'm not sure what I'm missing here but although both setups seem to be using the same model and parameters, the results are quite different.

**Comment by komninoschatzipapas** [comment]
@RandomGitUser321, thank you! Generating the seed on the CPU matched the image I was getting from ComfyUI.
